libgit2-sys/build.rs

```diff
@@ -31,33 +31,54 @@ fn main() {
     fs::create_dir_all(&include).unwrap();
 
     // Copy over all header files
-    cp_r("libgit2/include".as_ref(), &include);
+    cp_r("libgit2/include", &include);
 
     cfg.include(&include)
         .include("libgit2/src")
         .out_dir(dst.join("build"))
         .warnings(false);
 
     // Include all cross-platform C files
-    add_c_files(&mut cfg, "libgit2/src".as_ref());
-    add_c_files(&mut cfg, "libgit2/src/xdiff".as_ref());
+    add_c_files(&mut cfg, "libgit2/src");
+    add_c_files(&mut cfg, "libgit2/src/xdiff");
 
     // These are activated by features, but they're all unconditionally always
     // compiled apparently and have internal #define's to make sure they're
     // compiled correctly.
-    add_c_files(&mut cfg, "libgit2/src/transports".as_ref());
-    add_c_files(&mut cfg, "libgit2/src/streams".as_ref());
+    add_c_files(&mut cfg, "libgit2/src/transports");
+    add_c_files(&mut cfg, "libgit2/src/streams");
 
     // Always use bundled http-parser for now
     cfg.include("libgit2/deps/http-parser")
         .file("libgit2/deps/http-parser/http_parser.c");
 
-    // Always use bundled regex for now
-    cfg.include("libgit2/deps/regex")
-        .file("libgit2/deps/regex/regex.c");
+    // Use the included PCRE regex backend.
+    //
+    // Ideally these defines would be specific to the pcre files (or placed in
+    // a config.h), but since libgit2 already has a config.h used for other
+    // reasons, just define on the command-line for everything. Perhaps there
+    // is some way with cc to have different instructions per-file?
```

Comment:
Typically the way to go for this is to just use multiple `cc::Build` instances, but it's probably fine to do the same config for all of libgit2